js/enter_game.js

```diff
@@ -1,81 +1,134 @@
-  disableElment('1v1', false)
-  disableElment('2v2', true)
+    // display no results message
+    displayNoResultsMessage: false,
+    noResultsMessageText: 'Keine Ergebnisse zu "%s" gefunden.',
 
-  // disable inputs for second players
-  hideElment('playerA2', false) // textinput A
-  hideElment('playerB2', false) // textinput B
-  hideElment('checkboxA', false) // checkbox A
-  hideElment('checkboxB', false) // checkbox B
+    // data source
+    dataSource: 'select',
 
-  // check checkboxes
-  setCheckbox('checkboxA', true) // checkbox A
-  setCheckbox('checkboxB', true) // checkbox B
-}
+    // waiting time between each search
+    debounce: 0,
 
-function teamAHas2ndPlayer () { // eslint-disable-line no-unused-vars
-  // textinput A
-  if (isAhidden) {
-    hideElment('playerA2', false)
-    isAhidden = false
+    // custom placeholder text
+    placeholder: false,
+
+    // enable sortable
+    // requires jQuery UI
+    sortable: false,
+
+    // tabIndex
+    tabIndex: 0,
+
+    // allows empty values
+    allowEmptyValues: false,
+
+    // z-index
+    zIndexMargin: 500
+  })
+})
+
+function teamHas2ndPlayer (team, checkbox) { // eslint-disable-line no-unused-vars
+  let player
+  if (team === 'A') {
+    player = 'playerA2'
+  } else if (team === 'B') {
+    player = 'playerB2'
+  } else return
+
+  document.getElementById(player).disabled = !checkbox.checked
+  // enable/disable tokenize2 instance of player explicitly
+  const tok2p2 = tok2.find(tok2p => {
+    return tok2p.element[0].id === player
+  })
+  if (checkbox.checked) {
+    tok2p2.enable()
   } else {
-    hideElment('playerA2', true)
-    isAhidden = true
+    tok2p2.clear()
+    tok2p2.disable()
   }
 }
 
-function teamBHas2ndPlayer () { // eslint-disable-line no-unused-vars
-  // textinput B
-  if (isBhidden) {
-    hideElment('playerB2', false)
-    isBhidden = false
-  } else {
-    hideElment('playerB2', true)
-    isBhidden = true
+async function differentPlayersEntered (data) { // eslint-disable-line no-unused-vars
+  // create list of entered players
+  let playerIDList = [data.playerA1, data.playerA2, data.playerB1, data.playerB2]
+  // remove null from list of entered players -> list of actual players
+  playerIDList = playerIDList.filter(function (value, index, array) {
+    return value !== null
+  })
+  // create set of actual players -> remove duplicate entries (little reminiscence of The Exmatriculator)
+  const playerIDSet = new Set(playerIDList)
+  // entered players are all different, if list of actual players contains as much elements as set of actual players
+  return playerIDList.length === playerIDSet.size
+}
+
+async function validateUserInput (data) { // eslint-disable-line no-unused-vars
+  // check if enabled players are entered
+  if ((!document.getElementById('playerA2').disabled && !data.playerA2) || (!document.getElementById('playerB2').disabled && !data.playerB2)) {
+    console.log('Enter missing players or disable them!')
```

Comment:
Error messages like this must be displayed to the user, not (or not just) to the console